js/5etools-template.js

```diff
@@ -0,0 +1,502 @@
+				dropdown.append($("<option>", {
+					value: `${it.download_url}${d20plus.ut.getAntiCacheSuffix()}`,
+					text: `Homebrew: ${it.name.trim().replace(/\.json$/i, "")}`,
+				}));
+			});
+
+			dropdown.val(defaultUrl);
+			dropdown.change(function () {
+				$(inputFieldId).val(this.value);
+			});
+		}
+
+		function populateBasicDropdown (dropdownId, inputFieldId, defaultSel, brewProps, addForPlayers) {
+			function doPopulate (dropdownId, inputFieldId) {
+				const $sel = $(dropdownId);
+				const existingItems = !!$sel.find(`option`).length;
+				if (defaultSel) {
+					$(inputFieldId).val(defaultSel);
+					$sel.append($("<option>", {
+						value: defaultSel,
+						text: "Official Sources",
+					}));
+				}
+				if (!existingItems) {
+					$sel.append($("<option>", {
+						value: "",
+						text: "Custom",
+					}));
+				}
+
+				const dataList = [];
+				const seenPaths = new Set();
+				brewProps.forEach(prop => {
+					Object.entries(brewIndex[prop] || {})
+						.forEach(([path, dir]) => {
+							if (seenPaths.has(path)) return;
+							seenPaths.add(path);
+							dataList.push({
+								download_url: DataUtil.brew.getFileUrl(path),
+								path,
+								name: path.split("/").slice(1).join("/"),
+							});
+						});
+				});
+				dataList.sort((a, b) => SortUtil.ascSortLower(a.name, b.name)).forEach(it => {
+					$sel.append($("<option>", {
+						value: `${it.download_url}${d20plus.ut.getAntiCacheSuffix()}`,
+						text: `Homebrew: ${it.name.trim().replace(/\.json$/i, "")}`,
+					}));
+				});
+
+				$sel.val(defaultSel);
+				$sel.change(function () {
+					$(inputFieldId).val(this.value);
+				});
+			}
+
+			doPopulate(dropdownId, inputFieldId);
+			if (addForPlayers) doPopulate(`${dropdownId}-player`, `${inputFieldId}-player`);
+		}
+
+		const $body = $("body");
+
+		if (window.is_gm) {
+			const $wrpSettings = $(`#betteR20-settings`);
+
+			$wrpSettings.append(d20plus.template5e.settingsHtmlImportHeader);
+			$wrpSettings.append(d20plus.template5e.settingsHtmlSelector());
+			const $ptAdventures = $(d20plus.template5e.settingsHtmlPtAdventures);
+			$wrpSettings.append($ptAdventures);
+
+			IMPORT_CATEGORIES.forEach(ic => {
+				$wrpSettings.append(d20plus.template5e.getSettingsHTML(ic));
+			})
+			$ptAdventures.find(`.Vetools-module-tool-open`).click(() => d20plus.tool.get("MODULES").openFn());
+			$wrpSettings.append(d20plus.template5e.settingsHtmlPtImportFooter);
+
+			$("#bind-drop-locations").on(window.mousedowntype, d20plus.bindDropLocations);
+			$("#initiativewindow .characterlist").before(d20plus.template5e.initiativeHeaders);
+
+			d20plus.setTurnOrderTemplate();
+			d20.Campaign.initiativewindow.rebuildInitiativeList();
+			d20plus.hpAllowEdit();
+			d20.Campaign.initiativewindow.model.on("change:turnorder", function () {
+				d20plus.updateDifficulty();
+			});
+			d20plus.updateDifficulty();
+
+			const populateAdventuresDropdown = () => {
```

Comment:
as above; `d20plus.template5e._populateAdventuresDropdown`
or would this be better inlined, since it's only ever called once?